api/adapters/hydroshare.py

```diff
@@ -0,0 +1,260 @@
+        file_name: str
+        url: HttpUrl
+        size: int
+        content_type: str
+        logical_file_type: str
+        modified_time: datetime
+        checksum: str
+
+        def to_dataset_media_object(self):
+            media_object = schema.MediaObject.construct()
+            media_object.contentUrl = self.url
+            media_object.encodingFormat = self.content_type
+            media_object.contentSize = f"{self.size/1000.00} KB"
+            media_object.name = self.file_name
+            return media_object
+
+    class Relation(BaseModel):
+        type: str
+        value: str
+
+        def to_dataset_part_relation(self, relation_type: str):
+            relation = None
+            if relation_type == "IsPartOf" and self.type.endswith("is part of"):
+                relation = schema.IsPartOf.construct()
+            elif relation_type == "HasPart" and self.type.endswith("resource includes"):
+                relation = schema.HasPart.construct()
+            else:
+                return relation
+
+            description, url = self.value.rsplit(',', 1)
+            relation.description = description.strip()
+            relation.url = url.strip()
+            relation.name = self.value
+            return relation
+
+    class Rights(BaseModel):
+        statement: str
+        url: HttpUrl
+
+        def to_dataset_license(self):
+            _license = schema.License.construct()
+            _license.name = self.statement
+            _license.url = self.url
+            return _license
+
+    title: str
+    abstract: str
+    url: HttpUrl
+    identifier: HttpUrl
+    creators: List[Creator]
+    created: datetime
+    modified: datetime
+    subjects: Optional[List[str]]
+    language: str
+    rights: Rights
+    awards: Optional[List[Award]]
+    spatial_coverage: Optional[Union[SpatialCoverageBox, SpatialCoveragePoint]]
+    period_coverage: Optional[TemporalCoverage]
+    relations: Optional[List[Relation]]
+    citation: str
+    content_files: Optional[List[ContentFile]]
+
+    def to_dataset_creators(self):
+        creators = []
+        for creator in self.creators:
+            creators.append(creator.to_dataset_creator())
+        return creators
+
+    def to_dataset_grants(self):
+        grants = []
+        for award in self.awards:
+            grants.append(award.to_dataset_grant())
+        return grants
+
+    def to_dataset_media_objects(self):
+        media_objects = []
+        for content_file in self.content_files:
+            media_objects.append(content_file.to_dataset_media_object())
+        return media_objects
+
+    def to_dataset_part_relations(self, relation_type: str):
+        part_relations = []
+        for relation in self.relations:
+            part_relation = relation.to_dataset_part_relation(relation_type)
+            if part_relation:
+                part_relations.append(part_relation)
+        return part_relations
+
+    def to_catalog_dataset(self):
```

Comment:
I think some of this could be revisited to be more consistent in how the conversion logic is grouped.  
- Each part of provider is converted in this method
- Some are simple mappings
- Some are static values
- Some are mappings with small changes (wrapping the mapping in an array)
- Some call a method on this class
- Some call a method on a property on the class
- The method names are not consistent
- Some check if a property exists before setting a fields value

I think logic should be removed from this method.  Each field that isn't as a simple mapping or wrapping in an array should be refactored to a method. 

The method naming should be consistent/conventional.  I also think the methods should only be called from the class (not a method on a property).  The method implementation could call embedded property methods.  

I think the method naming convention should be `def to_dataset_{datasetPropertyName}(self):`